frontend/src/app/features/details/parcelDescriptions/parcelDescriptionsSlice.ts

```diff
@@ -1,163 +1,102 @@
 import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
 import { RequestStatus } from '../../../helpers/requests/status';
 import {
-  IParcelDescriptionDto,
   IParcelDescriptionResponseDto,
-} from './parcelDescriptionDto';
+  IParcelDescriptionsState,
+} from './parcelDescriptionsInterfaces';
 import { print } from 'graphql';
 import { getAxiosInstance } from '../../../helpers/utility';
 import { GRAPHQL } from '../../../helpers/endpoints';
-import { format } from 'date-fns';
 import { graphQLParcelDescriptionBySiteId } from '../../site/graphql/ParcelDescriptions';
 
-export interface IParcelDescriptionState {
-  siteId: number;
-  data: IParcelDescriptionDto[];
-  requestStatus: RequestStatus;
-  totalResults: number;
-  currentPage: number;
-  resultsPerPage: number;
-  searchParam: string;
-  sortBy: string;
-  sortByDir: string;
-  sortByInputValue: { [key: string]: any };
-}
-
-export interface IFetchParcelDescriptionParams {
-  siteId: number;
-  page: number;
-  pageSize: number;
-  searchParam: string;
-  sortBy: string;
-  sortByDir: string;
-  showPending: boolean;
-}
-
-const defaultValues = {
+export const initialParcelDescriptionsState: IParcelDescriptionsState = {
   siteId: 0,
   data: [],
-  requestStatus: RequestStatus.loading,
+  requestStatus: RequestStatus.idle,
   totalResults: 0,
   currentPage: 1,
   resultsPerPage: 5,
   searchParam: '',
   sortBy: 'id',
   sortByDir: 'ASC',
   sortByInputValue: {},
-};
-
-const initialState: IParcelDescriptionState = {
-  siteId: defaultValues.siteId,
-  data: defaultValues.data,
-  requestStatus: defaultValues.requestStatus,
-  totalResults: defaultValues.totalResults,
-  currentPage: defaultValues.currentPage,
-  resultsPerPage: defaultValues.resultsPerPage,
-  searchParam: defaultValues.searchParam,
-  sortBy: defaultValues.sortBy,
-  sortByDir: defaultValues.sortByDir,
-  sortByInputValue: defaultValues.sortByInputValue,
+  needsUpdate: true,
 };
 
 export const fetchParcelDescriptions = createAsyncThunk(
   'parcelDescriptions/fetchParcelDescriptions',
-  async (params: IFetchParcelDescriptionParams) => {
+  async (currentState: IParcelDescriptionsState) => {
```

Comment:
IMO, If the other comments are fixed and state is manage in React way. We dont need this logic here.